### PR TITLE
Fix fetch interface and fromInterface (Forward port #19676)

### PR DESCRIPTION
**Pull request by samuel-williams-da**
Currently the "Can fetch interface from V1 contract when V1 unvetted" test in "Interface.daml" is failing with the following error:
```
  com.digitalasset.daml.lf.engine.free.InterpretationError: Error: Unhandled Daml exception: Daml.Script.Internal.Questions.Testing:FailedCmd@d1c40430{ commandName = Daml.Script.Internal.Questions.Testing:CommandName@d1c40430{ getCommandName = "Submit" }, errorClassName = Daml.Script.Internal.Questions.Testing:ErrorClassName@d1c40430{ getErrorClassName = "StatusRuntimeException" }, errorMessage = Daml.Script.Internal.Questions.Testing:ErrorMessage@d1c40430{ getErrorMessage = "FAILED_PRECONDITION: INVALID_PRESCRIBED_DOMAIN_ID(9,5fae34f4): Cannot submit transaction to prescribed domain `mydomain::12207ce93435...` because: Some packages are not known to all informees on domain mydomain::12207ce93435...
```

The important part being `FAILED_PRECONDITION: INVALID_PRESCRIBED_DOMAIN_ID(9,5fae34f4): Cannot submit transaction to prescribed domain mydomain::12207ce93435... because: Some packages are not known to all informees on domain mydomain::12207ce93435.`

### CODE REVIEW

File: sdk/daml-script/test/daml/upgrades/Interfaces.daml [hunk 1]
```diff
@@ -153,7 +160,9 @@ main = tests
   , brokenOnIDELedger ("Multiple interface exercises use their respective package preference entries at command level", multipleInterfaceChoicePackagePreferenceCommand)
   , brokenOnIDELedger ("Multiple interface exercises use their respective package preference entries at choice body level", multipleInterfaceChoicePackagePreferenceUpdate)
   , broken ("queryInterfaceContractId uses highest version of view", queryInterfaceUpgrades)
-  , broken ("fetchFromInterface upgrades payload to match request", fetchFromInterfaceUpgrades)
+  , ("fetchFromInterface upgrades payload to match request", fetchFromInterfaceUpgrades)
+  -- IDE Ledger doesn't support unvetting
+  , broken ("Can fetch interface from V1 contract when V1 unvetted", fetchWithoutV1)
```
Comment (samuel-williams-da) [inline sdk/daml-script/test/daml/upgrades/Interfaces.daml:165]: On main-2.x, this works on Canton. Does seem to work on 3.x for some reason.

**Review by remyhaemmerle-da — APPROVED** [review]
Thanks

**Comment by samuel-williams-da** [comment]
Test failure seems to be related to https://github.com/DACH-NY/canton/issues/15220 not being fixed on 3x.
We will merge for now and mark this test as working once that ticket has been resolved

**Comment by remyhaemmerle-da** [comment]
Could you use a more descriptive title ?

**Comment by samuel-williams-da** [comment]
This fix doesn't account for transaction replays, thus is not sufficient.

**Pull request closed without merging**